torch/nn/quantized/dynamic/modules/rnn.py

```diff
@@ -58,45 +59,70 @@ def __init__(self, mode, input_size, hidden_size,
 
         packed_weights = []
         quantized_weights = []
-
+        orig_weights = []
         for layer in range(num_layers):
             for direction in range(num_directions):
                 layer_input_size = input_size if layer == 0 else hidden_size * num_directions
 
-                def process_weights(ihhh, layer, suffix, qweight, bias):
+                def process_weights(ihhh, layer, suffix, qweight, bias, dtype):
                     weight_name = 'weight_{}_l{}{}'.format(ihhh, layer, suffix)
                     bias_name = 'bias_{}_l{}{}'.format(ihhh, layer, suffix)
 
-                    # for each layer, for each direction we need to quantize and pack
-                    # weights and pack parameters in this order:
-                    #
-                    #   w_ih, w_hh, b_ih, b_hh
-                    packed_weight = \
-                        torch.ops.quantized.linear_prepack(qweight)
-                    params = [packed_weight, bias]
-                    pos_names = ['w', 'b']
-                    ret_name = ['{}_{}_l{}{}'.format(
-                        name, ihhh, layer, suffix) for name in pos_names]
-                    quantized_weights.append(qweight)
-                    packed_weights.append(ret_name[0])
-                    return params, ret_name
-
-                w_ih = torch._empty_affine_quantized(
-                    [gate_size, layer_input_size], scale=1, zero_point=0, dtype=torch.qint8)
-                w_hh = torch._empty_affine_quantized(
-                    [gate_size, hidden_size], scale=1, zero_point=0, dtype=torch.qint8)
-                b_ih = torch._empty_affine_quantized(
-                    [gate_size], scale=1, zero_point=0, dtype=torch.qint32)
-                # Second bias vector included for CuDNN compatibility. Only one
-                # bias vector is needed in standard definition.
-                b_hh = torch._empty_affine_quantized(
-                    [gate_size], scale=1, zero_point=0, dtype=torch.qint32)
+                    if dtype == torch.qint8:
+                        # for each layer, for each direction we need to quantize and pack
+                        # weights and pack parameters in this order:
+                        #
+                        #   w_ih, w_hh, b_ih, b_hh
+                        packed_weight = \
+                            torch.ops.quantized.linear_prepack(qweight)
```

Comment:
This should include bias, right? 

It's probably worth it to rebase on top of my change: https://github.com/pytorch/pytorch/pull/25877

Reply:
Have rebased on top of your Diff. Right: it should include the bias.